### PR TITLE
[MOV-4-TASK] added list of trailers to watch trailer screen

### PR DESCRIPTION
**Pull request by ArtemPolovin**

### CODE REVIEW

File: app/src/main/java/com/example/movies/ui/move_details/MovieDetailsFragment.kt [hunk 1]
```diff
@@ -116,18 +119,31 @@ class MovieDetailsFragment : Fragment() {
...
+                is ResponseResult.Failure -> {
+                    binding.btnWatchTrailer.visibility = GONE
+                }
+                else -> binding.btnWatchTrailer.visibility = GONE
+            }
         }
     }
 
+    private fun showWatchTrailerButton(trailerList: List<TrailerModel>) {
+        if (trailerList.isNotEmpty()) binding.btnWatchTrailer.visibility = VISIBLE
```
[the remark this answers is not in the document]
Comment (ArtemPolovin) [reply app/src/main/java/com/example/movies/ui/move_details/MovieDetailsFragment.kt:143]: Fixed

File: app/src/main/java/com/example/movies/ui/move_details/MovieDetailsFragment.kt [hunk 2]
```diff
@@ -116,18 +119,31 @@ class MovieDetailsFragment : Fragment() {
...
-    private fun showWatchTrailerButton(movieModel: MovieWithDetailsModel) {
-        movieModel.video?.let { video ->
-            if (video.isNotBlank()) binding.btnWatchTrailer.visibility = VISIBLE
-            else binding.btnWatchTrailer.visibility = GONE
+    private fun getTrailerList(movieId: Int) {
+        viewModel.fetchTrailersFromNetwork(movieId)
+        viewModel.trailerList.observe(viewLifecycleOwner) {
+
+            when (it) {
+                is ResponseResult.Success -> {
```
[the remark this answers is not in the document]
Comment (ArtemPolovin) [reply app/src/main/java/com/example/movies/ui/move_details/MovieDetailsFragment.kt:130]: I check it here in showWatchTrailerButton()

`    viewModel.trailerList.observe(viewLifecycleOwner) {

            when (it) {
                is ResponseResult.Success -> {
                    showWatchTrailerButton(it.data)`

File: app/src/main/java/com/example/movies/ui/watch_trailer/adapter/ViewPagerThumbnailsAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+import com.example.movies.R
+import com.example.movies.databinding.CellMovieHorizontalListBinding
+import com.example.movies.databinding.CellThumbnailBinding
+
+class ViewPagerThumbnailsAdapter :
+    ListAdapter<TrailerModel, ViewPagerThumbnailsAdapter.ThumbnailsViewHolder>(
+        ViewPagerThumbnailsDiffCallback()
+    ) {
+
+    var onItemClickListener: ((videoKey: String) -> Unit)? = null
```
[the remark this answers is not in the document]
Comment (ArtemPolovin) [reply app/src/main/java/com/example/movies/ui/watch_trailer/adapter/ViewPagerThumbnailsAdapter.kt:23]: Fixed

File: app/src/main/java/com/example/movies/ui/watch_trailer/adapter/ViewPagerThumbnailsAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+            CellThumbnailBinding.inflate(LayoutInflater.from(parent.context), parent, false)
+        return ThumbnailsViewHolder(binding,parent.context, onItemClickListener)
+
+    }
+
+    override fun onBindViewHolder(holder: ThumbnailsViewHolder, position: Int) {
+        val trailerModel = getItem(position)
+
+        holder.bind(trailerModel)
+        holder.onClickItem(trailerModel.videoKey)
```
Comment (startandroid) [inline app/src/main/java/com/example/movies/ui/watch_trailer/adapter/ViewPagerThumbnailsAdapter.kt:29]: onClickItem method can be called inside bind. 
Ideally, an adapter calls only holder.bind method. All the rest are done by holder, inside.

**Review by startandroid — APPROVED** [review]

**Review by startandroid — APPROVED** [review]

**Pull request merged**